src/extractor/guidance/roundabout_handler.cpp

```diff
@@ -131,43 +132,59 @@ void RoundaboutHandler::invalidateExitAgainstDirection(const NodeID from_nid,
     // *     | \░        | \          |    *
     // * <--'   \    <--'   v     <--'     *
     // *************************************
-    bool roundabout_entry_first = false;
-    auto invalidate_from = intersection.end(), invalidate_to = intersection.end();
-    for (auto road = intersection.begin(); road != intersection.end(); ++road)
+    const auto size = intersection.size();
+    auto intersection_entry = size, intersection_exit = size, road = size;
+    for (std::size_t index = 0; index < size; ++index)
     {
-        const auto &edge_data = node_based_graph.GetEdgeData(road->eid);
+        const auto &edge_data = node_based_graph.GetEdgeData(intersection[index].eid);
         if (edge_data.roundabout || edge_data.circular)
         {
             if (edge_data.reversed)
-            {
-                if (roundabout_entry_first)
-                { // invalidate turns in range exit..end
-                    invalidate_from = road + 1;
-                    invalidate_to = intersection.end();
-                }
-                else
-                { // invalidate turns in range begin..exit
-                    invalidate_from = intersection.begin() + 1;
-                    invalidate_to = road;
-                }
-            }
+                intersection_exit = index;
             else
-            {
-                roundabout_entry_first = true;
-            }
+                intersection_entry = index;
+        }
+        else if (road == size || intersection[road].angle == intersection[index].angle)
+        {
+            road = index;
         }
     }
 
+    OSRM_ASSERT(road != size, coordinates[from_nid]);
+    OSRM_ASSERT(intersection_entry != size, coordinates[from_nid]);
+    OSRM_ASSERT(intersection_exit != size, coordinates[from_nid]);
+    if (road == size || intersection_entry == size || intersection_exit == size)
+    { // ignore wrong intersections
```

Comment:
Right above you assert that all of these are unequal to size. Is this simply an additional save-guard, or am I missing something?